### ISSUE TITLE
PixelJunk Eden Hit Detection Issue

### ISSUE THREAD

**Issue by Peduls** [RPCS3/rpcs3#7971]
## Quick summary
When playing PixelJunk Eden the player phases through branches and objects that are normally supposed to be walked on making it impossible to play the game.
![screenshot-2020_04_06_16_19_03](https://user-images.githubusercontent.com/34750422/78610812-df5f3000-7822-11ea-8769-1ae000b79365.png)

## Details
On a real PS3 you swing from branches and stick to them in order to advance through each level. In RPCS3 the player cannot interact with these branches at all making it impossible to progress.

Operating System
	Windows 10 Enterprise 64-bit
CPU
	Intel Core i7 4790K @ 4.00GHz
	Haswell 22nm Technology
RAM
	16.0GB Dual-Channel DDR3 @ 799MHz (9-9-9-24)
Motherboard
	ASUSTeK COMPUTER INC. Z97-A (SOCKET 1150)
Graphics
	VG248 (1920x1080@144Hz)
	SAMSUNG (1920x1080@144Hz)
	4095MB NVIDIA GeForce GTX 1070 (EVGA)
Storage
	465GB Samsung SSD 850 EVO 500GB (SATA (SSD))
	2794GB TOSHIBA HDWD130 (SATA )
Optical Drives
	ASUS DRW-24B1ST j
Audio
	High Definition Audio Device

**Comment by Peduls** [comment]
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4441105/RPCS3.log.gz)

**Comment by AniLeo** [comment]
Test with Accurate XFloat

**Comment by Peduls** [comment]
Just tested with Accurate XFloat. The issue is still present.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4448056/RPCS3.log.gz)

**Comment by AniLeo** [comment]
Probably a CPU bug though not sure

**Comment by digitaldude555** [comment]
Still an issue.

**Comment by xrcipg** [comment]
Issue still present.

**Comment by Megamouse** [comment]
please test again

**Comment by xrcipg** [comment]
> please test again

ah wow.  I thought it had been fixed or some



thing  because of this message.  I tried it just now and still doesnt work.

https://use
![ps3](https://user-images.githubusercontent.com/73202874/223570197-21fbda22-f855-4358-8d2f-d73823f67ff3.png)
r-images.githubusercontent.com/73202874/223570155-968e80f3-098d-44f8-9eec-9fb18b7b4132.mp4

**Comment by Triticum0** [comment]
Retested with v0.0.30-15914-2a1af17f. No changes

**Comment by PlanetWilson** [comment]
v0.0.32-16453-fc92aef4 - still broken

**Comment by Tha14** [comment]
v0.0.32-16792-f2ef0966 - No change, still broken.

**Comment by UsahNaim** [comment]
Still absolutely broken

I thought it was due to the PixelJunk game version 2.00 (which IMHO broke the difficulty / playability) 
Also with Eden v1.05 : no luck

The issue is (for me) that the PC version is childlike-easy, so I 'need' the PS3 version to work.
No more 1 spectra at a time, but grabbing them all in one run - that's the only fun challenge of this game.. 
Without that it's just music 

Anyway, Good luck y'all

**Comment by matveydesign** [comment]
Still an issue BUT I noticed something interesting: If no input is detected at game start, it enters a demo mode with three players where two scenes play in succession. Collision isn't functional in the first scene but in the second scene, one of the players collides with a very thick pathway, bouncing off a couple of times but then still falling through as if the collision doesn’t fully register.

This might be worth investigating further to see if there’s something specific in the demo mode affecting collision logic. Attaching a video.

https://github.com/user-attachments/assets/028cad74-e60b-4f8b-b145-f5a70e5c6183

**Comment by UsahNaim** [comment]
>  PixelJunk.Eden.-.collision.issue.mp4
That is interesting.. Thanks! I hope someone can (and wants to) work his or her magic on this

I can only assume that, when fixed, this could be a huge bug-fix for more titles then PixelJunk Eden alone.

Good find!

**Comment by matveydesign** [comment]
@UsahNaim Good eye for the pollen haha! Same for me.

I found a 3rd scene with the same behaviour. Attaching a vid + log of running through all 3 demos.

https://github.com/user-attachments/assets/58eedfe4-4b12-4091-aa8f-619a612ede92
[RPCS3.log.zip](https://github.com/user-attachments/files/17708234/RPCS3.log.zip)

**Comment by nyakze** [comment]
RPCS3 Version: 0.0.34-17488-cefcaaf2 Alpha | master - no change, still unplayable

**Comment by linkmaster6** [comment]
0.0.35-17569 , game still unplayable

**Comment by nyakze** [comment]
0.0.37-18022-9c93ec0b
still the same issue

**Comment by nyakze** [comment]
RPCS3 Version: 0.0.38-18337-ec70c969 Alpha | master - no change, unplayable

**Comment by JamDaBam** [comment]
RPCS3 Version: 0.0.39-18705-0f85e912 Alpha | master - still persist, unplayable

**Comment by UsahNaim** [comment]
It's frustrating that it's still unplayable. Version 2 is out now {from what I've seen} there are checkpoints that you do not 'need' to grab, potentially giving us the original feeling again.
I guess I'll take it for a spin this week, I'll let you know (ONLY) if it somewhat resembles the original play-style ofc.

update:
Checkpoints are mandatory, they are the only way to get extra time in v2
Also, I have not (yet) been able to replay a level [at all] completed = completed
Other then collection pollen, it's a fundamentally different game

We just have to hope for a fix, or become a programmer ourselves

**Comment by nyakze** [comment]
> Version 2 is out now

Nah, Eden 2 is a mobile game (Eden Obscura) port, physics is nothing like the original. Controls are much more simplified.

RPCS3 Version: 0.0.39-18794-b30a44c1 Alpha | master - issue still persist. Can't stick to objects.

**Comment by nyakze** [comment]
0.0.40-18947-951d47c9 Alpha | master - no change, object physics' still not working, essentially unplayable

**Comment by nyakze** [comment]
0.0.41-19476-a4649475 Alpha | master - no change

**Comment by nyakze** [comment]
Here's a fresh log just in case... 

[RPCS3.log.gz](https://github.com/user-attachments/files/28977818/RPCS3.log.gz)

**Comment by nyakze** [comment]
Maybe a useful lead: PixelJunk Eden apparently uses a small GPU "picking buffer" for branch collision detection. It renders IDs into pixels and then reads that buffer back to detect what the player touches.

So maybe this is some RSX readback/sync issue rather than normal physics. Like the buffer gets rendered but the game reads stale/wrong data.

https://cedec.cesa.or.jp/2008/archives/file/pg12-1.pdf

**Comment by kd-11** [comment]
> Maybe a useful lead: PixelJunk Eden apparently uses a small GPU "picking buffer" for branch collision detection. It renders IDs into pixels and then reads that buffer back to detect what the player touches.
> 
> So maybe this is some RSX readback/sync issue rather than normal physics. Like the buffer gets rendered but the game reads stale/wrong data.
> 
> https://cedec.cesa.or.jp/2008/archives/file/pg12-1.pdf

Readback and writeback are fully implemented with options (WCB/RCB/WDB/RDB). I'll verify the PDF claims though just in case we are missing something obvious.

**Comment by UsahNaim** [comment]
Amazing find Nyakze!
If true and if this can be fixed (/implemented / changes with the correct settings) It might have a real impact on many titles!

Thank you kd-11,
I will try to fiddle with the options 👍

Update: No luck - See my log below
( After "E CFG: cfg::try_to_enum_value('Async Shader Recompiler','Shader Mode'): invalid enum or integer" I've reconfigured the settings to default + enabled the WCB/RCB/WDB/RDB settings )

[pixeljunk_eden.log](https://github.com/user-attachments/files/30782063/pixeljunk_eden.log)

Update 2:
Even on Strict Rendering Mode + precise GPU decoder, no async, no upscaling; no luck

**Comment by keikof070623-gif** [comment]
Might be worth retesting this game again, some synchronization fixes have been pushed since the last comment, perhaps they affected this game?

**Comment by Tha14** [comment]
> Might be worth retesting this game again, some synchronization fixes have been pushed since the last comment, perhaps they affected this game?

Tried it on 0.042-20043-b5a174e9 with multiple GPU and Advanced settings enabled but no bueno.
